hyracks/hyracks-dataflow-std/src/main/java/org/apache/hyracks/dataflow/std/structures/SerializableVector.java

```diff
@@ -1,13 +1,39 @@
 package org.apache.hyracks.dataflow.std.structures;
 
-public class SerializableVector implements ISerializableVector<IResetableComparable>{
+
+import org.apache.hyracks.api.context.IHyracksFrameMgrContext;
+
+import java.util.ArrayList;
+
+public class SerializableVector implements ISerializableVector<IResetableSerializable>{
+
+    ArrayList<byte[]> frames;
+    int recordSize;
+    int frameSize;
+    int recordPerFrame;
+    int sVectorSize;
+    int frameCount;
+
+    int lastPos;
+
+
+    private void init(int recordSize, int frameSize){
+        this.recordSize = recordSize;
+        this.frameSize = frameSize;
+        recordPerFrame = frameSize / recordSize;
+        sVectorSize = 0;
+        frameCount = 0;
+        lastPos = 0;
+    }
     /**
      * constructor
      * default frameSize is the same as system setting
      * @param recordSize
      */
-    public SerializableVector(IResetableComparableFactory factory, int recordSize){
-
+    public SerializableVector(IHyracksFrameMgrContext ctx, int recordSize){
+        frames = new ArrayList<>();
+        int frameSize = ctx.getInitialFrameSize();
+        init(recordSize, frameSize);
```

Comment:
should keep a field = ctx, we only use `ctx.allocateBuffer()` to allocate the memory unit.